bindings.go

```diff
@@ -85,10 +89,68 @@ func RawLibVersion(major *int, minor *int) (int, error) {
 	return int(n), getErr(err)
 }
 
+// RawDiscoverRUsageVersion finds the highest supported version of the rusage_info struct at runtime.
+//
+// Will return -1 on error.
+func RawDiscoverRUsageVersion() RUsageFlavor {
+	pid := C.getpid()
+	var v4 C.struct_rusage_info_v4
+	for i := 4; i >= 0; i-- {
+		_, err := C.pid_rusage(C.int(pid), C.int(i), (*C.rusage_info_t)(unsafe.Pointer(&v4)))
+		if err == nil {
+			return RUsageFlavor(i)
+		}
+		fmt.Println(err)
+	}
+	return -1
+}
+
 // RawPidRUsage is a low-level binding for proc_pid_rusage.
-func RawPidRUsage(pid Pid, flavor RUsageFlavor, buffer unsafe.Pointer) (int, error) {
-	n, err := C.pid_rusage(C.int(pid), C.int(flavor), (*C.rusage_info_t)(buffer))
-	return int(n), getErr(err)
+func RawPidRUsage(pid Pid) (*RUsage, int, error) {
```

Comment:
> I attempted to follow your approach of separating high/low-level functions, but it was difficult to achieve when copying the C struct to an equivalent Go struct.

FWIW, this was my first time using CGO, and I'm don't really have a problem with your approach, but what was the actual problem you faced "when copying the C struct to an equivalent Go struct"? It's been a while, but it _seems_ like you could call `RawPidRUsage` that accepts an `unsafe.Pointer` from `PidRUsage` the same way you call `C.pid_rusage` from `RawPidRUsage`. Curious more than anything :)